squarelet/organizations/fe_api/serializers.py

```diff
@@ -0,0 +1,129 @@
+# Django
+from django.db.models.expressions import F
+
+# Third Party
+import stripe
+from rest_framework import serializers, status
+from rest_framework.exceptions import APIException
+
+# Squarelet
+from squarelet.organizations.models import Invitation, Membership, Organization
+
+
+class OrganizationSerializer(serializers.ModelSerializer):
+    uuid = serializers.UUIDField(required=False)
+    # remove plan once all clients are updated to handle entitlements
+    plan = serializers.SerializerMethodField()
+    entitlements = serializers.SerializerMethodField()
+    merged = serializers.SlugRelatedField(read_only=True, slug_field="uuid")
+    member_count = serializers.SerializerMethodField()
+    admins = serializers.SerializerMethodField()
+    members = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Organization
+        fields = (
+            "id",
+            "uuid",
+            "admins",
+            "member_count",
+            "members",
+            "name",
+            "slug",
+            "plan",
+            "entitlements",
+            "max_users",
+            "individual",
+            "private",
+            "verified_journalist",
+            "updated_at",
+            "payment_failed",
+            "avatar_url",
+            "merged",
+        )
+
+    def get_plan(self, obj):
+        return obj.plan.slug if obj.plan else "free"
+
+    def get_entitlements(self, obj):
+        client = self.context.get("client")
+        if not client:
+            request = self.context.get("request")
+            if request and hasattr(request, "auth") and request.auth:
+                client = request.auth.client
+
+        if client:
+            return list(
+                client.entitlements.filter(plans__organizations=obj)
+                .annotate(update_on=F("plans__subscriptions__update_on"))
+                .values("name", "slug", "description", "resources", "update_on")
+            )
+        return []
+
+    def get_member_count(self, obj):
```

Comment:
This is potentially going to be slow, we may need to find a way to optimize this if we really need it

Reply:
Yeah, this is definitely a property we want to have.